datafusion/common/src/config.rs

```diff
@@ -614,6 +614,13 @@ config_namespace! {
         /// during aggregations, if possible
         pub enable_topk_aggregation: bool, default = true
 
+        /// When set to true attempts to push down dynamic filters generated by operators into the file scan phase.
```

Comment:
I think we could also extend non-scan operators with the dynamic filter pushdown?

It would be interesting to extend filters / aggregates / joins / etc. with the support of this as well...

Reply:
Would the idea be to prune hash table state, for example, if we knew some of the groups were no longer needed? 

I do think implementing more "late materialization" (aka turn on filter_pushdown) will help too